digital_benchmark/facebook_benchmark/data_provider.py

```diff
@@ -0,0 +1,44 @@
+import facebook as fb
+import requests
+
+from digital_benchmark.settings import facebook_graph_api_version, facebook_default_fields_for_page, facebook_default_fields_for_feed, facebook_default_fields_for_post, facebook_default_metrices_for_page_insights, facebook_default_metrices_for_post_insights
+
+class FacebookDataProvider:
+    def __init__(self, page_access_token, *args, **kwargs):
+        self.page_access_token = page_access_token
+        self.graph = fb.GraphAPI(access_token=page_access_token, version=facebook_graph_api_version)
+    
+    def get_page_details(self, fields=''):
+        if not fields:
+            fields = facebook_default_fields_for_page
+        page = self.graph.get_object(id='me', fields=fields)
+        return page
+    
+    def get_all_posts(self, fields=''):
+        if not fields:
```

Comment:
try not to change the value of the passed argument inside the function. There might be a use case in the future when we might need it. Make a new var